src/xtgeo/surface/regular_surface.py

```diff
@@ -1134,8 +1138,20 @@ def to_file(
         .. versionchanged:: 2.5 Added support for BytesIO
         .. versionchanged:: 2.13 Improved support for BytesIO
         .. versionchanged:: 2.14 Support for alias file name and return value
+        .. versionchanged:: 3.8 Add key ``error_if_near_empty``
         """
         logger.info("Export RegularSurface to file or memstream...")
+        if self.nactive is None or self.nactive < 4:
+            msg = (
+                f"Number of maps nodes are {self.nactive}. Exporting regular "
+                "surfaces with fewer than 4 nodes will not provide any "
+                "usable result. The map may also be not loaded if nodes are None."
+            )
+
+            if error_if_near_empty:
+                raise ValueError(msg)
```

Comment:
Question to reviewer(s): Is `ValueError` appropriate here?

Reply:
I think this is case where a `RuntimeError` actually might be more appropriate... but it doesn't fit cleanly enough into either category, in my opinion. Either seem fine to me.

ValueError

> Raised when an operation or function receives an argument that has the right type but an inappropriate value, and the situation is not described by a more precise exception such as [IndexError](https://docs.python.org/3/library/exceptions.html#IndexError).

RuntimeError

> Raised when an error is detected that doesn’t fall in any of the other categories. The associated value is a string indicating what precisely went wrong.